### PR TITLE
chore(sdk.v2): Deprecate dsl.ontology_artifacts, and migrate to using dsl.io_types

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
Currently we have mixed usage of `dsl.io_types` classes and `dsl.ontology_artifacts` classes. This causes problems, for instance, if we use `io_types.Model` in v2 lightweight function component, we actually build IR using `ontology_artifacts.Model`. 
This PR contains the following changes:
- move `SlicedClassificationMetrics` definition from `ontology_artifacts.py` to `io_types.py`
- delete `ontology_artifacts.py`, and move all references to `dsl.io_types`
- use `schemaTitle` exclusively (replaces `instanceSchema`)

This PR is also needed for the upcoming importer change, as we need to finalize which artifact classes to import.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by chensun** [comment]
@neuromage I see this change is breaking v2_compatible mode, as it's looking for `instance_schema`.

https://github.com/kubeflow/pipelines/blob/2c682299202ad019bb2b2bb2ac8903c72ab245cf/sdk/python/kfp/compiler/v2_compat.py#L126

Is this a scenario that we need the full `instance_schema`? or we can update it to use `schema_title`?

**Comment by neuromage** [comment]
> Is this a scenario that we need the full `instance_schema`? or we can update it to use `schema_title`?

PR #5458 actually fixes this and supports both `schema_title` and `instance_schema`

File: sdk/python/kfp/v2/google/aiplatform.py [hunk 1]
```diff
@@ -237,15 +235,14 @@ def _get_custom_job_op(
...
-  )
+      task_outputs=dummy_outputs)
 
 
 def custom_job(
     name: str,
     input_artifacts: Optional[Dict[str, dsl.PipelineParam]] = None,
     input_parameters: Optional[Dict[str, _ValueOrPipelineParam]] = None,
-    output_artifacts: Optional[Dict[str, Type[artifact.Artifact]]] = None,
+    output_artifacts: Optional[Dict[str, Type[io_types.Artifact]]] = None,
```
Comment (Ark-kun) [inline sdk/python/kfp/v2/google/aiplatform.py:245]: I think that "artifact" or "artifact_types" module name is better than "io_types". When I see "io_types" I think "File-based I/O", "URI-based I/O", "Network I/O", etc.

Comment (chensun) [reply sdk/python/kfp/v2/google/aiplatform.py:245]: @neuromage what do you think about the naming?

As we're deleting the current `dsl/artifact.py`, do we want to rename `io_types.py` to `artifact.py`?
Alternatively, we can promote `Artifacts`, `Model`, etc. to the dsl level (via `__init__.py`), so user would use `dsl.Artifact`, `dsl.Model`,  etc., and the file name becomes an implementation detail.

Comment (neuromage) [reply sdk/python/kfp/v2/google/aiplatform.py:245]: Let's keep this as is for now and revisit post-cleanup.

**Pull request merged**